packages/runtime/runtime-definitions/src/summary.ts

```diff
@@ -24,8 +27,19 @@ export interface ISummaryStats {
     unreferencedBlobSize: number;
 }
 
+/**
+ *  The summarization methods are recursively invoked during the summary calculation and will
+ *  compose the Summary Tree. At the same time, each component that is taking part of the summarization
+ *  will populate the tree information aggregation for its subtree through the ISummaryStats interface.
+ *  Any component that implements IChannelContext, IFluidDataStoreChannel or extends SharedObject
+ *  will be taking part of the summarization process.
```

Comment:
IMHO, this description may not be very relevant here. This is essentially describing the summarization process. It should rather say what this interface means. Basically, that this represents the summary tree for a node along with the statistics for that tree. As an example it could mention that for a data store this has the data for the data store along with a subtree for each of its DDS.

Reply:
Another important piece of information here may be its distinction from `ISummarizeResult` below.
Take a look at this issue which may help with some of the comments here - https://github.com/microsoft/FluidFramework/issues/4434

Reply:
Its also worth adding a comment for `ISummarizeInternalResult` as its not clear what it is and how it differs from `ISummarizeResult`